odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -193,10 +193,23 @@ def update_records(model, src, field_model='model', field_id='res_id'):
                 with mute_logger('odoo.sql_db'), self._cr.savepoint():
                     records.sudo().write({field_id: dst_partner.id})
                     records.env.flush_all()
-            except psycopg2.Error:
-                # updating fails, most likely due to a violated unique constraint
-                # keeping record with nonexistent partner_id is useless, better delete it
-                records.sudo().unlink()
+            except psycopg2.Error as e:
+                # updating fails, if the model of records is in allowed ones to delete in case
+                # of unique violation error then delete it, in case is not in allowed ones or
+                # postgresql error is not unique violation, then raise an error
+                model_names_to_delete = self.env['ir.config_parameter'].sudo().get_param("base.models_allowed_to_unlink_partner_merge")
+                model_names_to_delete = model_names_to_delete.split(",") if model_names_to_delete else []
+                if not model_names_to_delete or (records._name in model_names_to_delete and e.pgcode == psycopg2.errorcodes.UNIQUE_VIOLATION):
```

Comment:
This would be more readable if you split this line.

Reply:
Done